cpu/stm32_common/include/periph_cpu_common.h

```diff
@@ -153,6 +169,22 @@ typedef struct {
 } uart_conf_t;
 
 /**
+ * @brief   Structure for SPI configuration data
+ */
+typedef struct {
+    SPI_TypeDef *dev;       /**< SPI device base register address */
+    gpio_t mosi_pin;        /**< MOSI pin */
+    gpio_t miso_pin;        /**< MISO pin */
+    gpio_t sclk_pin;        /**< SCLK pin */
+    gpio_t cs_pin;          /**< HWCS pin, set to GPIO_UNDEF if not mapped */
+#ifndef CPU_FAM_STM32F1
+    gpio_af_t af;           /**< pin alternate function */
+#endif
+    uint32_t rccmask;       /**< bit in the RCC peripheral enable register */
+    uint8_t apbbus;         /**< APBx bus the device is connected to */
```

Comment:
control: with `uint8_t` I know for certain, that this field will only be 8bit, also I am no big fan in typing everything...